File: source_funcs/go_transcoder/CHECK_IF_A_NUMBER_IS_POWER_OF_ANOTHER_NUMBER_1/CHECK_IF_A_NUMBER_IS_POWER_OF_ANOTHER_NUMBER_1.go
package main

import "math"

func f_gold(x int, y int) bool {
	var (
		res1 int     = int(math.Log(float64(y)) / math.Log(float64(x)))
		res2 float64 = math.Log(float64(y)) / math.Log(float64(x))
	)
	return float64(res1) == res2
}

File: rust_funcs/c_transcoder/CHECK_IF_A_NUMBER_IS_POWER_OF_ANOTHER_NUMBER_1/CHECK_IF_A_NUMBER_IS_POWER_OF_ANOTHER_NUMBER_1.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(x: i32, y: i32) -> i32 {
    let res1 = (x as f32).log(y as f32).log(x as f32);
    let mut res2 = (y as f32).log(x as f32);
    res1 == res2 
}
